src/main/kotlin/org/rust/ide/docs/RustDocumentationProvider.kt

```diff
@@ -60,15 +60,29 @@ class RustDocumentationProvider : AbstractDocumentationProvider() {
         }
 
         val signatureStart = signatureStartElement?.startOffsetInParent ?: 0
-        val signatureEnd = block?.startOffsetInParent ?: textLength
+
+        // pick (in order) where clause, return type, or closing brace of the parameters
+        // if all else fails, drop down to the length of the current element
+        val functionElements = listOf(whereClause, retType, parameters)
+        val signatureEnd = functionElements
+                .filterNotNull()
```

Comment:
Maybe I should collapse these two into one line to make the intent clear (i.e. pick first non-null element in list)